### PR TITLE
added onHoverEnter and onHoverExit

### PR DESCRIPTION
**Pull request by lajbel**
Very high community requested!

This runs an event when is hovered, and other when is unhovered (this only are runned once, in diff with onHover)

added for KaboomCtx and BodyComp

Example:
```js
kaboom()

loadBean()

const bean = add([
    sprite("bean"),
    color(RED),
    pos(40, 80),
    area()
])

bean.onHoverEnter(() => {
    debug.log("hi")
})

bean.onHoverExit(() => {
    debug.log("bye")
})
```

Live Demo: https://replit.com/@lajbel/onHoverOnce?v=1

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -3655,15 +3664,27 @@ export interface AreaComp extends Comp {
 	 */
 	onClick(f: () => void): void,
 	/**
-	 * Register an event runs every frame when hovered.
+	 * Register an event runs once when hovered.
 	 *
 	 * @since v2000.1
```
Comment (slmjkdbtl) [inline src/types.ts:3669]: We may want to change this to `@since v3000.0`

File: src/types.ts [hunk 2]
```diff
@@ -730,14 +730,23 @@ export interface KaboomCtx {
...
+	/**
+	 * Register an event that runs every frame when game objs with certain tags are hovered (required to have area() component).
+	 *
+	 * @since v3000.0
+	 */
+	onHoverUpdate(tag: Tag, onHover: (a: GameObj) => void, onNotHover: (a: GameObj) => void): EventCanceller,
+	/**
+	 * Register an event that runs once when game objs with certain tags are unhovered (required to have area() component).
+	 *
+	 * @since v2001.0
```
Comment (slmjkdbtl) [inline src/types.ts:747]: Same for this, should be v3000.0

File: src/kaboom.ts [hunk 3]
```diff
@@ -4084,7 +4114,18 @@ export default (gopt: KaboomOpt = {}): KaboomCtx => {
 				})
 			},
 
-			onHover(onHover: () => void, onNotHover: () => void): EventCanceller {
+			onHover(action: () => void): EventCanceller {
+				return this.onUpdate(() => {
+					if(this.isHovering()) {
+						if (this.hoverStarted) return
+						this.hoverStarted = true
+						this.hoverEnded = false
```
Comment (slmjkdbtl) [inline src/kaboom.ts:4122]: These states should be initialized at the top of the object, and have type definition in types.ts under `AreaComp`

**Review by slmjkdbtl — APPROVED** [review]
Looks good, thanks! Just some small suggestions

**Comment by lajbel** [comment]
omg! i didn't know you could mention people in commits, sorry Since!

**Comment by slmjkdbtl** [comment]
Mr. Since is very popular here

**Pull request merged**